[python] Claude Model Parser

Ankush-lastmile opened the pull request:
[python] Claude Model Parser









Model Parser for Claude on AWS Bedrock



Claude on Bedrock can be called with the
- `anthropic_bedrock` python library
- AWS' `Boto3` Library
- Chose to use the `anthropic_bedrock` library because it has good abstractions and type hints

Claude on Bedrock only supports Text Completions, turn style [prompts are not supported](https://docs.anthropic.com/claude/reference/claude-on-amazon-bedrock#:~:text=Messages%20in%20Amazon%20Bedrock)

- Added Claude model parser as a core parser.
- added dependency on `anthropic_bedrock`



## Testplan

using prompt schema from diff ontop

https://github.com/lastmile-ai/aiconfig/assets/141073967/29c1faa7-7d13-412f-8606-9ad556eb1c52

---
[//]: # (BEGIN SAPLING FOOTER)
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/lastmile-ai/aiconfig/pull/1039).
* #1050
* __->__ #1039

Ankush-lastmile commented on python/src/aiconfig/Config.py:
```diff
@@ -53,6 +54,7 @@
     DefaultAnyscaleEndpointParser("AnyscaleEndpoint")
 )
 ModelParserRegistry.register_model_parser(GeminiModelParser("gemini-pro"), ["gemini-pro"])
+ModelParserRegistry.register_model_parser(ClaudeBedrockModelParser(), ["Claude"])
```
Need to figure out model vs model provider but for now just registering under "Claude" as well

Ankush-lastmile commented on python/src/aiconfig/default_parsers/claude.py:
```diff
@@ -0,0 +1,297 @@
+        # See https://docs.anthropic.com/claude/docs/introduction-to-prompt-design#human--assistant-formatting
+        completion_data[
+            "prompt"
+        ] = f"{HUMAN_PROMPT} {resolved_prompt}{AI_PROMPT}"
```
Unsure if its the best idea to manipulate the prompt inside the model parser

Ankush-lastmile commented on python/src/aiconfig/default_parsers/claude.py:
```diff
@@ -0,0 +1,297 @@
+
+        completion_data["stream"] = stream
+
+        response = self.client.completions.create(**completion_data)  # type: ignore (pyright doesn't understand response object)
```
API call will throw If aws credentials are not set. I removed mine locally and tested this manually.

rossdanlm commented on python/src/aiconfig/default_parsers/claude.py:
```diff
@@ -0,0 +1,297 @@
+            if isinstance(output_data, str):
+                return output_data
+
+            # Claude outputs should only ever be string
+            # format so shouldn't get here, but just being safe
```
That's what I'm talking about

rossdanlm reviewed (approved):
Nice, big ups

Ankush-lastmile commented:
- Removed "Claude" from default model parser registry as a Model Id, just have parser id for now. Will need to revist.
- addressed comments, most of them were nits.
